core/src/autogluon/core/scheduler/mo_asha.py

```diff
@@ -0,0 +1,107 @@
+import logging
+import copy
+import numpy as np
+
+from .hyperband import HyperbandScheduler, HyperbandBracketManager
+from .mo_asha_promotion import EPS_NET, NSGA_II
+from ..utils.default_arguments import filter_by_key
+from ..utils.mo_hbo_utils import prepare_sign_vector
+from ..utils.mo_hbo_utils import retrieve_pareto_front
+
+
+__all__ = ['MOASHAScheduler']
+
+logger = logging.getLogger(__name__)
+
+_ARGUMENT_KEYS = {
+    'objectives',
+}
+
+
+class MOASHAScheduler(HyperbandScheduler):
+    r"""Implements different variants of asynchronous multi-objective Hyperband
+    by extending the standard HyperbandScheduler using a multi-objective
+    version of ASHA. This version internally ranks candidates by first sorting
+    them into different Pareto frontiers and then spaces them out evenly based
+    on euclidean distance.
+
+    Parameters
+    ----------
+    objectives : dict
+        Dictionary with the names of objectives of interest. The corresponding
+        values are allowed to be either "MAX" or "MIN" and indicate if an
+        objective is to be maximized or minimized.
+
+    Examples
+    --------
+    examples/mo_hpo/mo_asha.ipynb
+    """
+
+    def __init__(self, train_fn, **kwargs):
```

Comment:
Code you inserted in HyperbandScheduler must be moved here. Try to avoid copy&paste by using private methods?